redact/redact.go

```diff
@@ -0,0 +1,59 @@
+package redact
+
+import (
+	"net/http"
+	"strings"
+)
+
+var SensitiveFields = []string{
+	"authorization",
+	"cookie",
+	"set-cookie",
+
+	"access_token",
+	"refresh_token",
+	"id_token",
+	"accesstoken",
+	"refreshtoken",
+	"idtoken",
+
+	"password",
+	"secret",
+}
+
+func IsSensitive(key string) bool {
+	for _, sensitiveKey := range SensitiveFields {
+		if strings.Contains(strings.ToLower(key), sensitiveKey) {
+			return true
+		}
+	}
+	return false
+}
+
+func Map(data map[string]interface{}) map[string]interface{} {
+	redactedData := map[string]interface{}{}
```

Comment:
Since we know ahead of time, may as well plan for it:
```suggestion
	redactedData := make(map[string]interface{}, len(data))
```